### DISCUSSION TITLE
Issue with outputting string variables containing equal sign (=)

### DISCUSSION THREAD

**Discussion by HarryHung** [nextflow-io/nextflow#3552] [Q&A] [answered]
I am not sure whether it is a bug of Nextflow or I am doing it wrong.

If I use output type `env` and the string variable contains a equal sign `=`, only the part before the `=` will be output, even though the whole string is captured in `.command.env`

Example code:
```
process TEST {
    output:
    env OUTPUT_VAR

    shell:
    '''
    OUTPUT_VAR='PART1=PART2'
    '''
}


workflow {
    TEST().view()
}
```

Terminal output:
```
PART1
```

Expected output:
```
PART1=PART2
```

The `.command.env` is as following, showing that the whole variable was successfully captured. But without any quote, it seems the information after `=` is lost after further usage of this variable
```
OUTPUT_VAR=PART1=PART2
```

I don't think this is the expected behaviour, as standard bash can have string variables containing `=`
```
OUTPUT_VAR='PART1=PART2'
echo $OUTPUT_VAR

PART1=PART2
``` 

I am grateful for any idea or suggestion. Or let me know if I should submit an issue. 

Thank you!

**Comment by pditommaso** [comment, marked as answer]
Likely is related to this https://github.com/nextflow-io/nextflow/blob/c492ad227d77b729b3c951767395b83a95d0ff6f/modules/nextflow/src/main/groovy/nextflow/processor/TaskProcessor.groovy#L1496-L1504

**Comment by HarryHung** [comment]
Thanks, looks like it. 
It feels like a bug to me, should I report it?

**Reply by pditommaso** [reply]
Yes, please

**Comment by HarryHung** [comment]
Issue created:
https://github.com/nextflow-io/nextflow/issues/3553

**Reply by mribeirodantas** [reply]
Can't find this issue 🤔

**Reply by pditommaso** [reply]
This https://github.com/nextflow-io/nextflow/issues/3553 ?

**Reply by HarryHung** [reply]
It seems the issue disappear for me as well. I did not delete it though...

**Reply by HarryHung** [reply]
Somehow, it returned.

**Reply by mribeirodantas** [reply]
Yeah, I can see it now. That was weird 😅
